### ISSUE TITLE
Make setuptools install routine work on ReadTheDocs

### ISSUE THREAD

**Issue by ezdac** [raiden-network/microraiden#438]
`python setup.py install --force` fails to install microraiden into a virtualenv.
This is crucial for having autodoc features in the deployed version of the documentation on http://mircoraiden.readthedocs.io .
The build inside a virtual environment there fails due to a depend
Unfortunately, this is how RTD installs projects:

```
update_docs_from_vcs(version)
if exists('setup.py'):
    run('python setup.py install')
if project.requirements_file:
    run('pip install -r %s' % project.requirements_file)
build_docs(version=version)
copy_files(artifact_dir)
```

So we have to rely on the setuptools install.
It seems that there is a problem of finding the `flask` package on the index?

I cannot confirm locally, because there ia different error occurs for `python setup.py install --force` on Mac OS.

Below is the output on RTD:

```
Installed /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_RESTful-0.3.6-py3.5.egg
Searching for flask_httpauth
Reading https://pypi.python.org/simple/flask_httpauth/
Downloading https://pypi.python.org/packages/6f/73/2a8da636f18ed580507a34483cb4fe26ef8d466909972029ebb3605415d2/Flask-HTTPAuth-3.2.3.tar.gz#md5=20f31ed9eb28ca04de57bece707c63d7
Best match: Flask-HTTPAuth 3.2.3
Processing Flask-HTTPAuth-3.2.3.tar.gz
Writing /tmp/easy_install-uwrbax4y/Flask-HTTPAuth-3.2.3/setup.cfg
Running Flask-HTTPAuth-3.2.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-uwrbax4y/Flask-HTTPAuth-3.2.3/egg-dist-tmp-gtizp2i_
creating /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_HTTPAuth-3.2.3-py3.5.egg
Extracting Flask_HTTPAuth-3.2.3-py3.5.egg to /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages
Adding Flask-HTTPAuth 3.2.3 to easy-install.pth file

Installed /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_HTTPAuth-3.2.3-py3.5.egg
Searching for flask
Downloading https://pypi.python.org/packages/20/f1/14a62bba209ae189e5c5fa33d5e0b7a4b5969488fa71fd3b8b323860bfc8/Flask-RESTful-0.3.6.tar.gz#md5=8243ca101185b130d8fa529cbdbdbddd
Best match: Flask RESTful-0.3.6
Processing Flask-RESTful-0.3.6.tar.gz
Writing /tmp/easy_install-p_aac1kc/Flask-RESTful-0.3.6/setup.cfg
Running Flask-RESTful-0.3.6/setup.py -q bdist_egg --dist-dir /tmp/easy_install-p_aac1kc/Flask-RESTful-0.3.6/egg-dist-tmp-3ezydrm6
warning: no files found matching 'CHANGES'
warning: no previously-included files matching '*.pyc' found under directory 'docs'
warning: no previously-included files matching '*.pyo' found under directory 'docs'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
warning: no previously-included files matching '*.pyo' found under directory 'tests'
warning: no previously-included files matching '*.pyc' found under directory 'examples'
warning: no previously-included files matching '*.pyo' found under directory 'examples'
no previously-included directories found matching 'docs/_build'
no previously-included directories found matching 'docs/_themes'
removing '/home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_RESTful-0.3.6-py3.5.egg' (and everything under it)
creating /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_RESTful-0.3.6-py3.5.egg
Extracting Flask_RESTful-0.3.6-py3.5.egg to /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages
Flask-RESTful 0.3.6 is already the active version in easy-install.pth

Installed /home/docs/checkouts/readthedocs.org/user_builds/microraiden/envs/docs-develop/lib/python3.5/site-packages/Flask_RESTful-0.3.6-py3.5.egg
error: The 'flask' distribution was not found and is required by flask-httpauth, microraiden
```

**Comment by ezdac** [comment]
For some obscure reason setuptools doesn't find the Flask distribution like this (as seen in the log above)
I found out that this can be solved by specifiying a specific Flask version, e.g.

`Flask>=0.12`

I will create a PR that incorporates this.

**Comment by ezdac** [comment]
Fixed in #454
